### PR TITLE
Feature/avs 464 add granular push consent to sdk

### PR DESCRIPTION
**Pull request by adrianaudienceview**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -5,23 +5,13 @@ apply plugin: 'com.google.gms.google-services'
 apply plugin: 'com.google.firebase.appdistribution'
 
 private String getCommitMessages() {
-    def note = "Build number:${getBuildNumber()} \n"
+    def note = "Build number:1 \n"
```
Comment (rossp-av) [inline app/build.gradle:8]: How come we're changing this?

File: sdk/src/main/java/com/realifetech/sdk/communicate/domain/PushConsentRepository.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import javax.inject.Inject
+
+class PushConsentRepository @Inject constructor(private val dataSource: PushConsentDataSource) {
+    private lateinit var result: Result<Boolean>
+
+    fun getNotificationConsents(): List<NotificationConsent> {
+        var notificationConsentList = emptyList<NotificationConsent>()
+        dataSource.getNotificationConsents { error, response ->
+            error?.let {
+                Log.e("PushConsentRepository", "Get notification consents Error: ${it.message}")
```
Comment (rossp-av) [inline sdk/src/main/java/com/realifetech/sdk/communicate/domain/PushConsentRepository.kt:17]: Do we just want to log, or do something else?

File: sdk/src/main/java/com/realifetech/sdk/communicate/domain/PushConsentRepository.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+        return notificationConsentList
+    }
+
+    fun getMyNotificationConsents(): List<DeviceNotificationConsent> {
+        var deviceNotificationConsentList = emptyList<DeviceNotificationConsent>()
+        dataSource.getMyNotificationConsents { error, response ->
+            error?.let {
+                Log.e("PushConsentRepository", "Get my notification consents Error: ${it.message}")
```
Comment (rossp-av) [inline sdk/src/main/java/com/realifetech/sdk/communicate/domain/PushConsentRepository.kt:30]: Same question

File: sdk/src/test/java/com/realifetech/sdk/general/GeneralTest.kt [hunk 4]
```diff
@@ -49,7 +49,7 @@ class GeneralTest {
     @Test
     fun `register Device results with success`() {
         every { sdkInitializationPrecondition.checkContextInitialized() } returns Unit
-        every { deviceRepository.registerDevice() } returns Result.Success(deviceRegisterResponse)
+        every { deviceRepository.registerDevice() } returns Result.Success(true)
```
Comment (You-Hsuan) [inline sdk/src/test/java/com/realifetech/sdk/general/GeneralTest.kt:52]: May I know why we need to change the tests for register device please? (same question for other test files). Thank you.

**Pull request merged**